core/src/main/java/com/netflix/iceberg/hadoop/HadoopTableOperations.java

```diff
@@ -47,13 +47,15 @@
 
   private final Configuration conf;
   private final Path location;
+  private final FileSystem metadataFs;
   private TableMetadata currentMetadata = null;
   private Integer version = null;
   private boolean shouldRefresh = true;
 
   HadoopTableOperations(Path location, Configuration conf) {
     this.conf = conf;
     this.location = location;
+    this.metadataFs = Util.getFS(location, conf);
```

Comment:
These changes look fine to me, along with the package-private changes to HadoopInputFile and HadoopOutputFile.